### ISSUE TITLE
readMSData-like constructor 

### ISSUE THREAD

**Issue by lgatto** [rformassspectrometry/Spectra#41]
There is currently no `readMSData` constructor as we have in `MSnbase`. We don't strictly need one, but I think it is extremely convenient and can simplify many standard use cases and abstract the relatively complex concept of backend (that isn't relevant at all for most users). So what would you think of having a `readMSData`-like function that would transparently use an appropriate backend depending on the file type:
- `mz[X]ML` and `netcdf`: use `MsBackendMzR`
- `mgf`: use `MsBackendDataFrame`
- possibly also `raw`: use `MsBackendRawDiag` 
- ...

comments from @rformassspectrometry/core

**Comment by jorainer** [comment]
I at first also wanted to implement sort of a `readMSData` function, but, based on the feedback from @sgibb, implemented the `Spectra,character` constructor: this constructor has an additional argument `source` that allows to specify a `MsBackend` that is supposed to know what to do with the `character` provided. So, the default is:

```r
Spectra("file.mzML", source = MsBackendMzR(), backend = MsBackendDataFrame())
```

The function passes the `character` (`"file.mzML"`) to the `backendInitialize` method of the `source` and that imports or handles the data. After *source* backend initialization, the backend is changed to `backend` (in the example above to the `MsBackendDataFrame`). If `backend = MsBackendMzR()` is used the backend would obviously not changed.
This would work straight out of the box for e.g. the `MsBackendRawDiag`, if their `backendInitialize` supports initialization from a file.
For other things, like e.g. an HMDB MS2 spectrum xml file there could be a `MsBackendHmdbXml` backend that knows how to read/handle xml files from HMDB.

If we provide a good documentation of this concept users will start loving it. It is much more flexible than a classical *read the data* approach (if really needed we could build that on top).

**Comment by lgatto** [comment]
We don't need to do this now (not a priority at this point), but requiring a new user to read documentation and understand non-trivial concepts to get started is a significant barrier. If we can anticipate 90% of the use cases with a simple *readMyData* function that does *the right thing*, I think that would be useful. But let's wait before tacking this.

**Comment by jorainer** [comment]
But shouldn't that function go into the `MsExperiment` package? Most people will want to read their experiment and read e.g. mzML files along with sample annotations. This might also ease the transition from `MSnbase` to `MsExperiment`.

**Comment by lgatto** [comment]
Yes, that's an excellent suggestion! Transferring this issue to `MSnExperiment`.
